ts/client/src/accounts/perp.ts

```diff
@@ -409,22 +410,27 @@ export class PerpMarket {
     direction: 'negative' | 'positive',
     count = 2,
   ): Promise<{ account: MangoAccount; settleablePnl: I80F48 }[]> {
-    let accs = (await client.getAllMangoAccounts(group))
-      .filter((acc) =>
-        // need a perp position in this market
-        acc.perpPositionExistsForMarket(this),
-      )
-      .map((acc) => {
-        return {
-          account: acc,
-          settleablePnl: acc
-            .perpActive()
-            .find((pp) => pp.marketIndex === this.perpMarketIndex)!
-            .getPnl(this),
-        };
-      });
-
-    accs = accs
+    const filteredAccounts = (await client.getAllMangoAccounts(group)).filter(
+      (acc) => acc.perpPositionExistsForMarket(this),
+    );
+    filteredAccounts.forEach((acc) =>
+      acc
+        .perpActive()
+        .find((pp) => pp.marketIndex === this.perpMarketIndex)!
+        .updateSettleLimit(this),
+    );
+
+    let accountsWithSettleablePnl = filteredAccounts.map((acc) => {
+      return {
```

Comment:
this would read more easily to me if the `find()!` and `updateSettleLimit()` were done in this closure, just before the return statement; later it'd be `settleablePnl: perpPosition.getSettleablePnl(this)`